# Prediction of Movie Revenue

Dataset from Kaggle : **"Full TMDB Movies Dataset 2024"** by *ASANICZKA*  
Source: https:www.kaggle.com/datasets/asaniczka/tmdb-movies-dataset-2023-930k-movies

### Essential Libraries

In [823]:
import numpy as np
import pandas as pd
import seaborn as sb
import matplotlib.pyplot as plt # we only need pyplot
sb.set() # set the default Seaborn style for graphics
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import OneHotEncoder, StandardScaler
from sklearn.compose import ColumnTransformer
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_absolute_error, mean_squared_error, r2_score

### Importing Dataset & Listing Variables

In [824]:
tmdbdata = pd.read_csv('TMDB_movie_dataset_v11.csv')
tmdbdata.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1024235 entries, 0 to 1024234
Data columns (total 24 columns):
 #   Column                Non-Null Count    Dtype  
---  ------                --------------    -----  
 0   id                    1024235 non-null  int64  
 1   title                 1024223 non-null  object 
 2   vote_average          1024235 non-null  float64
 3   vote_count            1024235 non-null  int64  
 4   status                1024235 non-null  object 
 5   release_date          897532 non-null   object 
 6   revenue               1024235 non-null  int64  
 7   runtime               1024235 non-null  int64  
 8   adult                 1024235 non-null  bool   
 9   backdrop_path         286248 non-null   object 
 10  budget                1024235 non-null  int64  
 11  homepage              110690 non-null   object 
 12  imdb_id               577196 non-null   object 
 13  original_language     1024235 non-null  object 
 14  original_title        1024223 non-

### Sieving columns

These will be the columns we work with.

In [825]:
sievedtmdbdata = pd.DataFrame(tmdbdata[['id', 'title','status', 'revenue', 'vote_average', 'vote_count','runtime', 'adult', 'budget', 'original_language', 'popularity', 'genres', 'production_companies', 'production_countries', 'spoken_languages']])
sievedtmdbdata.shape
#sievedtmdbdata.head()

(1024235, 15)

### Column Description (need to add for votecount)
1. id: Unique identifier for each movie. (type: int)
2. title: Title of the movie. (type: str)
3. status: The status of the movie (e.g., Released, Rumored, Post Production, etc.). (type: str)
4. revenue: Total revenue generated by the movie. (type: int)
5. vote_average: Average vote or rating given by viewers. (type: float)
6. runtime: Duration of the movie in minutes. (type: int)
7. adult: Indicates if the movie is suitable only for adult audiences. (type: bool)
8. budget: Budget allocated for the movie. (type: int)
9. original_language: Original language in which the movie was produced. (type: str)
10. popularity: Popularity score of the movie. (type: float)
11. genres: List of genres the movie belongs to. (type: str)
12. production_companies: List of production companies involved in the movie. (type: str)
13. production_countries: List of countries involved in the movie production. (type: str)
14. spoken_languages: List of languages spoken in the movie. (type: str)

### Cleaning Data

- Conversion of all int 0 values to NaN
- Dropping Nan:
        - Within the dataset there exists movies where the Revenue=0, Runtime=0, Budget=0,Vote_count=0.... These circumstances are unlikely to be valid since majority of movies have non-zero values for these variables.

In [826]:
sievedtmdbdata.replace(0, np.nan, inplace=True)
sievedtmdbdata

id                         title    status       revenue  \
0          27205                     Inception  Released  8.255328e+08   
1         157336                  Interstellar  Released  7.017292e+08   
2            155               The Dark Knight  Released  1.004558e+09   
3          19995                        Avatar  Released  2.923706e+09   
4          24428                  The Avengers  Released  1.518816e+09   
...          ...                           ...       ...           ...   
1024230   692340    Zoltan Kaszas: Modern Male  Released           NaN   
1024231   692341  Händel / Mozart: Der Messias  Released           NaN   
1024232   692342         Jak se naučit švédsky  Released           NaN   
1024233   692345                    Bumbhaiyya  Released           NaN   
1024234  1276193                    Young Love  Released           NaN   

         vote_average  vote_count  runtime  adult       budget  \
0               8.364     34495.0    148.0  False  160000000.0   
1               8.417     32571.0    169.0  False  165000000.0   
2               8.512     30619.0    152.0  False  185000000.0   
3               7.573     29815.0    162.0  False  237000000.0   
4               7.710     29166.0    143.0  False  220000000.0   
...               ...         ...      ...    ...          ...   
1024230           NaN         NaN     65.0  False          NaN   
1024231           NaN         NaN    135.0  False          NaN   
1024232           NaN         NaN      NaN  False          NaN   
1024233           NaN         NaN      5.0  False          NaN   
1024234           NaN         NaN      NaN  False          NaN   

        original_language  popularity  \
0                      en      83.952   
1                      en     140.241   
2                      en     130.643   
3                      en      79.932   
4                      en      98.082   
...                   ...         ...   
1024230                en       0.863   
1024231                de       0.600   
1024232                cs       0.600   
1024233                en       0.600   
1024234                en         NaN   

                                              genres  \
0                 Action, Science Fiction, Adventure   
1                  Adventure, Drama, Science Fiction   
2                     Drama, Action, Crime, Thriller   
3        Action, Adventure, Fantasy, Science Fiction   
4                 Science Fiction, Action, Adventure   
...                                              ...   
1024230                                       Comedy   
1024231                                        Music   
1024232                                       Comedy   
1024233                                          NaN   
1024234                                          NaN   

                                      production_companies  \
0        Legendary Pictures, Syncopy, Warner Bros. Pict...   
1        Legendary Pictures, Syncopy, Lynda Obst Produc...   
2        DC Comics, Legendary Pictures, Syncopy, Isobel...   
3        Dune Entertainment, Lightstorm Entertainment, ...   
4                                           Marvel Studios   
...                                                    ...   
1024230                                                NaN   
1024231                                  3sat, ORF, Unitel   
1024232  Československá televize Praha, Filmové studio ...   
1024233                                                NaN   
1024234                                                NaN   

                             production_countries  \
0        United Kingdom, United States of America   
1        United Kingdom, United States of America   
2        United Kingdom, United States of America   
3        United States of America, United Kingdom   
4                        United States of America   
...                                           ...   
1024230                  United States of America 

### Dropping all those rows with na value in those specific columns for numerical and categorical visualisation

In [827]:
sievedtmdbdata.dropna(subset=['revenue', 'budget', 'runtime', 'vote_average', 'popularity', 'genres', 'vote_count', 'original_language'], inplace=True)
sievedtmdbdata.shape

(10233, 15)

- After cleaning, there are 10233 rows of existing movies that have complete information in the required fields for both numerical and categorical analysis

In [828]:
sievedtmdbdata

id             title    status       revenue  vote_average  \
0        27205         Inception  Released  8.255328e+08         8.364   
1       157336      Interstellar  Released  7.017292e+08         8.417   
2          155   The Dark Knight  Released  1.004558e+09         8.512   
3        19995            Avatar  Released  2.923706e+09         7.573   
4        24428      The Avengers  Released  1.518816e+09         7.710   
...        ...               ...       ...           ...           ...   
346034  592192  WWR Adios Aurora  Released  1.122000e+03        10.000   
346035  309625   Winter Carnival  Released  4.742860e+05         6.000   
346215  280402      Freaky Night  Released  1.000000e+03         4.500   
346490  402714             Sming  Released  8.903800e+04         6.000   
347099  878495    The Foul Truth  Released  2.709600e+04        10.000   

        vote_count  runtime  adult       budget original_language  popularity  \
0          34495.0    148.0  False  160000000.0                en      83.952   
1          32571.0    169.0  False  165000000.0                en     140.241   
2          30619.0    152.0  False  185000000.0                en     130.643   
3          29815.0    162.0  False  237000000.0                en      79.932   
4          29166.0    143.0  False  220000000.0                en      98.082   
...            ...      ...    ...          ...               ...         ...   
346034         1.0    110.0  False       8665.0                en       1.127   
346035         1.0    105.0  False     412640.0                en       0.636   
346215         1.0     15.0  False        500.0                en       0.600   
346490         1.0    105.0  False   15000000.0                th       1.405   
347099         1.0     28.0  False       4065.0                en       0.600   

                                             genres  \
0                Action, Science Fiction, Adventure   
1                 Adventure, Drama, Science Fiction   
2                    Drama, Action, Crime, Thriller   
3       Action, Adventure, Fantasy, Science Fiction   
4                Science Fiction, Action, Adventure   
...                                             ...   
346034                                Drama, Action   
346035                       Romance, Comedy, Drama   
346215                                       Horror   
346490                        Drama, Action, Horror   
347099                                     Thriller   

                                     production_companies  \
0       Legendary Pictures, Syncopy, Warner Bros. Pict...   
1       Legendary Pictures, Syncopy, Lynda Obst Produc...   
2       DC Comics, Legendary Pictures, Syncopy, Isobel...   
3       Dune Entertainment, Lightstorm Entertainment, ...   
4                                          Marvel Studios   
...                                                   ...   
346034                       Women's Wrestling Revolution   
346035                          Walter Wanger Productions   
346215                                     Mystery Forest   
346490                          Fast Time Motion Pictures   
347099                             Priyakanta Productions   

                            production_countries  \
0       United Kingdom, United States of America   
1       United Kingdom, United States of America   
2       United Kingdom, United States of America   
3       United States of America, United Kingdom   
4                       United States of America   
...                                          ...   
346034                  United States of America   
346035                  United States of America   
346215                                    Norway   
346490                                  Thailand   
347099                                       NaN   

                          spoken_languages  
0       English, French, Japanese, Swahili  
1                                  Englis